provider/adapters/src/fixed.rs

```diff
@@ -0,0 +1,99 @@
+// This file is part of ICU4X. For terms of use, please see the file
+// called LICENSE at the top level of the ICU4X source tree
+// (online at: https://github.com/unicode-org/icu4x/blob/main/LICENSE ).
+
+//! Data provider always serving the same struct.
+
+use icu_provider::any::MaybeSendSync;
+use icu_provider::prelude::*;
+
+/// A data provider that returns clones of a fixed type-erased payload.
+///
+/// [`FixedProvider`] implements [`AnyProvider`], so it can be used in
+/// `*_with_any_provider` constructors across ICU4X.
+/// # Examples
+///
+/// ```
+/// use icu_provider::hello_world::*;
+/// use icu_provider::prelude::*;
+/// use icu_provider_adapters::any_payload::FixedProvider;
+/// use std::borrow::Cow;
+/// use writeable::assert_writeable_eq;
+///
+/// let provider =
+///     FixedProvider::from_static::<HelloWorldV1Marker>(&HelloWorldV1 {
+///         message: Cow::Borrowed("custom hello world"),
+///     });
+///
+/// // Check that it works:
+/// let formatter = HelloWorldFormatter::try_new_with_any_provider(
+///     &provider,
+///     &Default::default(),
+/// )
+/// .expect("marker matches");
+/// assert_writeable_eq!(formatter.format(), "custom hello world");
+///
+/// # struct DummyMarker;
+/// # impl DynamicDataMarker for DummyMarker {
+/// #     type DataStruct = <HelloWorldV1Marker as DynamicDataMarker>::DataStruct;
+/// # }
+/// # impl DataMarker for DummyMarker {
+/// #     const INFO: DataMarkerInfo = DataMarkerInfo::from_path(icu_provider::marker::data_marker_path!("dummy@1"));
+/// # }
+/// // Requests for invalid markers get MissingDataMarker
+/// assert!(matches!(
+///     provider.load_any(DummyMarker::INFO, Default::default()),
+///     Err(DataError {
+///         kind: DataErrorKind::MarkerNotFound,
+///         ..
+///     })
+/// ))
+/// ```
+#[derive(Debug)]
+#[allow(clippy::exhaustive_structs)] // this type is stable
+pub struct FixedProvider<M: DataMarker> {
+    /// The [`AnyPayload`] to return on matching requests.
+    data: M::DataStruct,
```

Comment:
Suggestion: I think this should just be a `DataPayload<M>`, because then it can be constructed from a static ref or a yoke.

Note that it was previously an `AnyPayload`, which wraps a `DataPayload`.

Probably make it a public field.

Reply:
That's a good idea, then we can keep the old constructors